spec/features/user_sign_up_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.feature 'User sign ups', type: :feature do
+  let!(:user) { create(:user) }
+
+  background do
+    visit root_path
+    click_on 'Sign up'
+  end
+
+  scenario 'with valid values' do
+    fill_in 'Email', with: 'jvargas18@gmail.com'
+    fill_in 'Password', with: '123456'
+    fill_in 'Password confirmation', with: '123456'
+    click_on 'Sign up'
+    expect(page).to have_content('Welcome! You have signed up successfully.')
+    expect(current_path).to eq(root_path)
+  end
+
+  scenario 'with email already taken' do
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    fill_in 'Password confirmation', with: user.password
+    click_on 'Sign up'
+    expect(page).to have_content('Email has already been taken')
+    expect(current_path).to eq(user_registration_path)
+  end
+
+  scenario 'with a very short password' do
+    fill_in 'Email', with: 'jvargas18@gmail.com'
+    fill_in 'Password', with: '12345'
+    fill_in 'Password confirmation', with: '12345'
+    click_on 'Sign up'
+    expect(page).to have_content('Password is too short')
+    expect(current_path).to eq(user_registration_path)
```

Comment:
here you could take advantage of faker to generate this data: 
```suggestion
    short_password = Faker::Internet.password(min_length: 3, max_length: 5)
    user = build(:user, password: short_password)
    fill_in 'Email', with: user.email
    fill_in 'Password', with: user.password
    fill_in 'Password confirmation', with: user.password
    click_on 'Sign up'
    expect(page).to have_content('Password is too short')
    expect(current_path).to eq(user_registration_path)
```